examples/blasius.py

```diff
@@ -0,0 +1,554 @@
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import os
+import logging
+import numpy as np
+
+from logpyle import IntervalTimer, set_dt
+
+from grudge.shortcuts import make_visualizer, compiled_lsrk45_step
+from grudge.dof_desc import DISCR_TAG_QUAD, DD_VOLUME_ALL
+
+from mirgecom.discretization import create_discretization_collection
+from mirgecom.navierstokes import ns_operator
+from mirgecom.simutil import (
+    check_step,
+    get_sim_timestep,
+    write_visfile,
+    check_naninf_local,
+    distribute_mesh,
+    global_reduce
+)
+from mirgecom.restart import write_restart_file
+from mirgecom.io import make_init_message
+from mirgecom.mpi import mpi_entry_point
+from mirgecom.steppers import advance_state
+from mirgecom.boundary import (
+    PressureOutflowBoundary,
+    AdiabaticSlipBoundary,
+    IsothermalWallBoundary,
+    LinearizedInflowBoundary,
+    LinearizedOutflowBoundary,
+)
+from mirgecom.utils import force_evaluation
+from mirgecom.fluid import make_conserved
+from mirgecom.eos import IdealSingleGas
+from mirgecom.gas_model import GasModel, make_fluid_state
+from mirgecom.logging_quantities import (
+    initialize_logmgr,
+    logmgr_add_cl_device_info, logmgr_set_time,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class Initializer():
+
+    def __init__(self, dim, pressure, velocity, species_mass_fractions):
+        self._dim = dim
+        self._velocity = velocity
+
+    def __call__(self, x_vec, eos):
+
+        actx = x_vec[0].array_context
+
+        temp_y = 1.0 + actx.np.tanh(1.0/0.01*x_vec[1])
+        temp_x = 1.0 + 0.5*(1.0 - actx.np.tanh(1.0/0.01*(x_vec[0]+0.02)))
+        temperature = actx.np.maximum(temp_y, temp_x)
+        pressure = 1.0
+        mass = pressure/(eos.gas_const()*temperature)
+        velocity = self._velocity
+
+        energy = pressure/(eos.gamma() - 1.0) + 0.5*mass*np.dot(velocity, velocity)
+
+        return make_conserved(dim=self._dim,
+                              mass=mass,
+                              energy=energy,
+                              momentum=mass*velocity)
```

Comment:
Add to initializers?

Reply:
This is not a Blasius solution per-se. It is just a smoothed field to avoid crashing right away due to hot flow/cold wall. Thus, I don't think it should be in the initializers.